src2/com/piseth/java/school/service/PersonServiceImpl2.java

```diff
@@ -0,0 +1,134 @@
+package com.piseth.java.school.service;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.piseth.java.school.model.Person2;
+
+public class PersonServiceImpl2 implements PersonService2{
+	
+	PreparedStatement pstm;
+	ResultSet rs;
+	
+	@Override
+	public Map<String, Integer> countNumberOfPeopleByGender(Connection conn) {		
+		try {
+			pstm = conn.prepareStatement("SELECT gender, count(*) FROM persons GROUP BY gender");
+			rs = pstm.executeQuery();
+			Map<String, Integer> mapGender = new LinkedHashMap<>();
+			while (rs.next()) {
+				mapGender.putIfAbsent(rs.getString(1), rs.getInt(2));
+			}
+			return mapGender;
+			
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return null;
+		
+	}
+
+	@Override
+	public List<Person2> findByNumberOfPetMoreThan(int minNum, Connection conn) {
+		try {
+			pstm = conn.prepareStatement("SELECT * FROM persons WHERE CARDINALITY(pet) > ?");
+			pstm.setInt(1, minNum);
+			rs = pstm.executeQuery();
+			ArrayList<Person2> personList = new ArrayList<>();
+			while (rs.next()) {
+				Person2 person = new Person2(rs.getString(1), rs.getString(2), rs.getInt(3), rs.getArray(4));
```

Comment:
You use this statement multiple time in different method. You can create a function for it to reuse code.